CompletionException: java.lang.NullPointerException

<!--
Please make sure to first check our common issues page before creating a new issue:
https://github.com/Imprex-Development/Orebfuscator/wiki/Common-Issues
-->

**Please fill out the following information:**

* Server Version: spigot 1.12.2 (final build)
* Orebfuscator Version: orebfuscator-plugin-5.2.4.jar
* Protocollib Version: ProtocolLib.jar 4.7.0

**Describe the bug**
The plugin is working absolutely fine. However, after I join my normal world. My console get spammed like this:
```
[01:45:39] [ofc-task-worker-1/WARN]: java.util.concurrent.CompletionException: java.lang.NullPointerException
[01:45:39] [ofc-task-worker-1/WARN]: 	at java.util.concurrent.CompletableFuture.encodeRelay(CompletableFuture.java:326)
[01:45:39] [ofc-task-worker-1/WARN]: 	at java.util.concurrent.CompletableFuture.completeRelay(CompletableFuture.java:338)
[01:45:39] [ofc-task-worker-1/WARN]: 	at java.util.concurrent.CompletableFuture.uniRelay(CompletableFuture.java:925)
[01:45:39] [ofc-task-worker-1/WARN]: 	at java.util.concurrent.CompletableFuture$UniRelay.tryFire(CompletableFuture.java:913)
[01:45:39] [ofc-task-worker-1/WARN]: 	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
[01:45:39] [ofc-task-worker-1/WARN]: 	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
[01:45:39] [ofc-task-worker-1/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationRequest.completeExceptionally(ObfuscationRequest.java:73)
[01:45:39] [ofc-task-worker-1/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationTask.completeExceptionally(ObfuscationTask.java:44)
[01:45:39] [ofc-task-worker-1/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.process(ObfuscationProcessor.java:103)
[01:45:39] [ofc-task-worker-1/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationTaskWorker.run(ObfuscationTaskWorker.java:39)
[01:45:39] [ofc-task-worker-1/WARN]: 	at java.lang.Thread.run(Thread.java:748)
[01:45:39] [ofc-task-worker-1/WARN]: Caused by: java.lang.NullPointerException
[01:45:39] [ofc-task-worker-1/WARN]: 	at net.imprex.orebfuscator.nms.v1_12_R1.ReadOnlyChunkWrapper.getBlockState(ReadOnlyChunkWrapper.java:16)
[01:45:39] [ofc-task-worker-1/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationTask.getBlockState(ObfuscationTask.java:49)
[01:45:39] [ofc-task-worker-1/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.isAdjacentBlockOccluding(ObfuscationProcessor.java:135)
[01:45:39] [ofc-task-worker-1/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.shouldObfuscate(ObfuscationProcessor.java:123)
[01:45:39] [ofc-task-worker-1/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.process(ObfuscationProcessor.java:74)
[01:45:39] [ofc-task-worker-1/WARN]: 	... 2 more
[01:45:39] [ofc-task-worker-0/WARN]: java.util.concurrent.CompletionException: java.lang.NullPointerException
[01:45:39] [ofc-task-worker-0/WARN]: 	at java.util.concurrent.CompletableFuture.encodeRelay(CompletableFuture.java:326)
[01:45:39] [ofc-task-worker-0/WARN]: 	at java.util.concurrent.CompletableFuture.completeRelay(CompletableFuture.java:338)
[01:45:39] [ofc-task-worker-0/WARN]: 	at java.util.concurrent.CompletableFuture.uniRelay(CompletableFuture.java:925)
[01:45:39] [ofc-task-worker-0/WARN]: 	at java.util.concurrent.CompletableFuture$UniRelay.tryFire(CompletableFuture.java:913)
[01:45:39] [ofc-task-worker-0/WARN]: 	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
[01:45:39] [ofc-task-worker-0/WARN]: 	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
[01:45:39] [ofc-task-worker-0/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationRequest.completeExceptionally(ObfuscationRequest.java:73)
[01:45:39] [ofc-task-worker-0/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationTask.completeExceptionally(ObfuscationTask.java:44)
[01:45:39] [ofc-task-worker-0/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.process(ObfuscationProcessor.java:103)
[01:45:39] [ofc-task-worker-0/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationTaskWorker.run(ObfuscationTaskWorker.java:39)
[01:45:39] [ofc-task-worker-0/WARN]: 	at java.lang.Thread.run(Thread.java:748)
[01:45:39] [ofc-task-worker-0/WARN]: Caused by: java.lang.NullPointerException
[01:45:39] [ofc-task-worker-0/WARN]: 	at net.imprex.orebfuscator.nms.v1_12_R1.ReadOnlyChunkWrapper.getBlockState(ReadOnlyChunkWrapper.java:16)
[01:45:39] [ofc-task-worker-0/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationTask.getBlockState(ObfuscationTask.java:49)
[01:45:39] [ofc-task-worker-0/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.isAdjacentBlockOccluding(ObfuscationProcessor.java:135)
[01:45:39] [ofc-task-worker-0/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.shouldObfuscate(ObfuscationProcessor.java:123)
[01:45:39] [ofc-task-worker-0/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.process(ObfuscationProcessor.java:74)
[01:45:39] [ofc-task-worker-0/WARN]: 	... 2 more
[01:45:39] [ofc-task-worker-5/WARN]: java.util.concurrent.CompletionException: java.lang.NullPointerException
[01:45:39] [ofc-task-worker-5/WARN]: 	at java.util.concurrent.CompletableFuture.encodeRelay(CompletableFuture.java:326)
[01:45:39] [ofc-task-worker-5/WARN]: 	at java.util.concurrent.CompletableFuture.completeRelay(CompletableFuture.java:338)
[01:45:39] [ofc-task-worker-5/WARN]: 	at java.util.concurrent.CompletableFuture.uniRelay(CompletableFuture.java:925)
[01:45:39] [ofc-task-worker-5/WARN]: 	at java.util.concurrent.CompletableFuture$UniRelay.tryFire(CompletableFuture.java:913)
[01:45:39] [ofc-task-worker-5/WARN]: 	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
[01:45:39] [ofc-task-worker-5/WARN]: 	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
[01:45:39] [ofc-task-worker-5/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationRequest.completeExceptionally(ObfuscationRequest.java:73)
[01:45:39] [ofc-task-worker-5/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationTask.completeExceptionally(ObfuscationTask.java:44)
[01:45:39] [ofc-task-worker-5/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.process(ObfuscationProcessor.java:103)
[01:45:39] [ofc-task-worker-5/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationTaskWorker.run(ObfuscationTaskWorker.java:39)
[01:45:39] [ofc-task-worker-5/WARN]: 	at java.lang.Thread.run(Thread.java:748)
[01:45:39] [ofc-task-worker-5/WARN]: Caused by: java.lang.NullPointerException
[01:45:39] [ofc-task-worker-5/WARN]: 	at net.imprex.orebfuscator.nms.v1_12_R1.ReadOnlyChunkWrapper.getBlockState(ReadOnlyChunkWrapper.java:16)
[01:45:39] [ofc-task-worker-5/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationTask.getBlockState(ObfuscationTask.java:49)
[01:45:39] [ofc-task-worker-5/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.isAdjacentBlockOccluding(ObfuscationProcessor.java:135)
[01:45:39] [ofc-task-worker-5/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.shouldObfuscate(ObfuscationProcessor.java:123)
[01:45:39] [ofc-task-worker-5/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.process(ObfuscationProcessor.java:74)
[01:45:39] [ofc-task-worker-5/WARN]: 	... 2 more
[01:45:39] [ofc-task-worker-4/WARN]: java.util.concurrent.CompletionException: java.lang.NullPointerException
[01:45:39] [ofc-task-worker-4/WARN]: 	at java.util.concurrent.CompletableFuture.encodeRelay(CompletableFuture.java:326)
[01:45:39] [ofc-task-worker-4/WARN]: 	at java.util.concurrent.CompletableFuture.completeRelay(CompletableFuture.java:338)
[01:45:39] [ofc-task-worker-4/WARN]: 	at java.util.concurrent.CompletableFuture.uniRelay(CompletableFuture.java:925)
[01:45:39] [ofc-task-worker-4/WARN]: 	at java.util.concurrent.CompletableFuture$UniRelay.tryFire(CompletableFuture.java:913)
[01:45:39] [ofc-task-worker-4/WARN]: 	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
[01:45:39] [ofc-task-worker-4/WARN]: 	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
[01:45:39] [ofc-task-worker-4/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationRequest.completeExceptionally(ObfuscationRequest.java:73)
[01:45:39] [ofc-task-worker-4/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationTask.completeExceptionally(ObfuscationTask.java:44)
[01:45:39] [ofc-task-worker-4/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.process(ObfuscationProcessor.java:103)
[01:45:39] [ofc-task-worker-4/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationTaskWorker.run(ObfuscationTaskWorker.java:39)
[01:45:39] [ofc-task-worker-4/WARN]: 	at java.lang.Thread.run(Thread.java:748)
[01:45:39] [ofc-task-worker-4/WARN]: Caused by: java.lang.NullPointerException
[01:45:39] [ofc-task-worker-4/WARN]: 	at net.imprex.orebfuscator.nms.v1_12_R1.ReadOnlyChunkWrapper.getBlockState(ReadOnlyChunkWrapper.java:16)
[01:45:39] [ofc-task-worker-4/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationTask.getBlockState(ObfuscationTask.java:49)
[01:45:39] [ofc-task-worker-4/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.isAdjacentBlockOccluding(ObfuscationProcessor.java:135)
[01:45:39] [ofc-task-worker-4/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.shouldObfuscate(ObfuscationProcessor.java:123)
[01:45:39] [ofc-task-worker-4/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.process(ObfuscationProcessor.java:74)
[01:45:39] [ofc-task-worker-4/WARN]: 	... 2 more
[01:45:39] [ofc-task-worker-2/WARN]: java.util.concurrent.CompletionException: java.lang.NullPointerException
[01:45:39] [ofc-task-worker-2/WARN]: 	at java.util.concurrent.CompletableFuture.encodeRelay(CompletableFuture.java:326)
[01:45:39] [ofc-task-worker-2/WARN]: 	at java.util.concurrent.CompletableFuture.completeRelay(CompletableFuture.java:338)
[01:45:39] [ofc-task-worker-2/WARN]: 	at java.util.concurrent.CompletableFuture.uniRelay(CompletableFuture.java:925)
[01:45:39] [ofc-task-worker-2/WARN]: 	at java.util.concurrent.CompletableFuture$UniRelay.tryFire(CompletableFuture.java:913)
[01:45:39] [ofc-task-worker-2/WARN]: 	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
[01:45:39] [ofc-task-worker-2/WARN]: 	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
[01:45:39] [ofc-task-worker-2/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationRequest.completeExceptionally(ObfuscationRequest.java:73)
[01:45:39] [ofc-task-worker-2/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationTask.completeExceptionally(ObfuscationTask.java:44)
[01:45:39] [ofc-task-worker-2/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.process(ObfuscationProcessor.java:103)
[01:45:39] [ofc-task-worker-2/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationTaskWorker.run(ObfuscationTaskWorker.java:39)
[01:45:39] [ofc-task-worker-2/WARN]: 	at java.lang.Thread.run(Thread.java:748)
[01:45:39] [ofc-task-worker-2/WARN]: Caused by: java.lang.NullPointerException
[01:45:39] [ofc-task-worker-2/WARN]: 	at net.imprex.orebfuscator.nms.v1_12_R1.ReadOnlyChunkWrapper.getBlockState(ReadOnlyChunkWrapper.java:16)
[01:45:39] [ofc-task-worker-2/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationTask.getBlockState(ObfuscationTask.java:49)
[01:45:39] [ofc-task-worker-2/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.isAdjacentBlockOccluding(ObfuscationProcessor.java:135)
[01:45:39] [ofc-task-worker-2/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.shouldObfuscate(ObfuscationProcessor.java:123)
[01:45:39] [ofc-task-worker-2/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.process(ObfuscationProcessor.java:74)
[01:45:39] [ofc-task-worker-2/WARN]: 	... 2 more
[01:45:39] [ofc-task-worker-1/WARN]: java.util.concurrent.CompletionException: java.lang.NullPointerException
[01:45:39] [ofc-task-worker-1/WARN]: 	at java.util.concurrent.CompletableFuture.encodeRelay(CompletableFuture.java:326)
[01:45:39] [ofc-task-worker-1/WARN]: 	at java.util.concurrent.CompletableFuture.completeRelay(CompletableFuture.java:338)
[01:45:39] [ofc-task-worker-1/WARN]: 	at java.util.concurrent.CompletableFuture.uniRelay(CompletableFuture.java:925)
[01:45:39] [ofc-task-worker-1/WARN]: 	at java.util.concurrent.CompletableFuture$UniRelay.tryFire(CompletableFuture.java:913)
[01:45:39] [ofc-task-worker-1/WARN]: 	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
[01:45:39] [ofc-task-worker-1/WARN]: 	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
[01:45:39] [ofc-task-worker-1/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationRequest.completeExceptionally(ObfuscationRequest.java:73)
[01:45:39] [ofc-task-worker-1/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationTask.completeExceptionally(ObfuscationTask.java:44)
[01:45:39] [ofc-task-worker-1/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.process(ObfuscationProcessor.java:103)
[01:45:39] [ofc-task-worker-1/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationTaskWorker.run(ObfuscationTaskWorker.java:39)
[01:45:39] [ofc-task-worker-1/WARN]: 	at java.lang.Thread.run(Thread.java:748)
[01:45:39] [ofc-task-worker-1/WARN]: Caused by: java.lang.NullPointerException
[01:45:39] [ofc-task-worker-3/WARN]: java.util.concurrent.CompletionException: java.lang.NullPointerException
[01:45:39] [ofc-task-worker-3/WARN]: 	at java.util.concurrent.CompletableFuture.encodeRelay(CompletableFuture.java:326)
[01:45:39] [ofc-task-worker-3/WARN]: 	at java.util.concurrent.CompletableFuture.completeRelay(CompletableFuture.java:338)
[01:45:39] [ofc-task-worker-3/WARN]: 	at java.util.concurrent.CompletableFuture.uniRelay(CompletableFuture.java:925)
[01:45:39] [ofc-task-worker-3/WARN]: 	at java.util.concurrent.CompletableFuture$UniRelay.tryFire(CompletableFuture.java:913)
[01:45:39] [ofc-task-worker-3/WARN]: 	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
[01:45:39] [ofc-task-worker-3/WARN]: 	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
[01:45:39] [ofc-task-worker-3/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationRequest.completeExceptionally(ObfuscationRequest.java:73)
[01:45:39] [ofc-task-worker-3/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationTask.completeExceptionally(ObfuscationTask.java:44)
[01:45:39] [ofc-task-worker-3/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.process(ObfuscationProcessor.java:103)
[01:45:39] [ofc-task-worker-3/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationTaskWorker.run(ObfuscationTaskWorker.java:39)
[01:45:39] [ofc-task-worker-3/WARN]: 	at java.lang.Thread.run(Thread.java:748)
[01:45:39] [ofc-task-worker-3/WARN]: Caused by: java.lang.NullPointerException
[01:45:39] [ofc-task-worker-1/WARN]: java.util.concurrent.CompletionException: java.lang.NullPointerException
[01:45:39] [ofc-task-worker-1/WARN]: 	at java.util.concurrent.CompletableFuture.encodeRelay(CompletableFuture.java:326)
[01:45:39] [ofc-task-worker-1/WARN]: 	at java.util.concurrent.CompletableFuture.completeRelay(CompletableFuture.java:338)
[01:45:39] [ofc-task-worker-1/WARN]: 	at java.util.concurrent.CompletableFuture.uniRelay(CompletableFuture.java:925)
[01:45:39] [ofc-task-worker-1/WARN]: 	at java.util.concurrent.CompletableFuture$UniRelay.tryFire(CompletableFuture.java:913)
[01:45:39] [ofc-task-worker-1/WARN]: 	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
[01:45:39] [ofc-task-worker-1/WARN]: 	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990)
[01:45:39] [ofc-task-worker-1/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationRequest.completeExceptionally(ObfuscationRequest.java:73)
[01:45:39] [ofc-task-worker-1/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationTask.completeExceptionally(ObfuscationTask.java:44)
[01:45:39] [ofc-task-worker-1/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationProcessor.process(ObfuscationProcessor.java:103)
[01:45:39] [ofc-task-worker-1/WARN]: 	at net.imprex.orebfuscator.obfuscation.ObfuscationTaskWorker.run(ObfuscationTaskWorker.java:39)
[01:45:39] [ofc-task-worker-1/WARN]: 	at java.lang.Thread.run(Thread.java:748)
```
...

This are actually two different errors but they are both java.lang.NullPointerException. You should copy my cutout into a texteditor to have a clear look of the error.

Hope you can fix it. The plugin is working totally fine but this console errors are really annoying.
Thank you!

--- a/orebfuscator-nms/orebfuscator-nms-v1_10_R1/src/main/java/net/imprex/orebfuscator/nms/v1_10_R1/NmsManager.java
+++ b/orebfuscator-nms/orebfuscator-nms-v1_10_R1/src/main/java/net/imprex/orebfuscator/nms/v1_10_R1/NmsManager.java
@@ -49,7 +49,7 @@ public class NmsManager extends AbstractNmsManager {
 
 		if (isChunkLoaded(worldServer, x >> 4, z >> 4) || loadChunk) {
 			// will load chunk if not loaded already
-			Chunk chunk = chunkProviderServer.getOrLoadChunkAt(x >> 4, z >> 4);
+			Chunk chunk = chunkProviderServer.getChunkAt(x >> 4, z >> 4);
 			return chunk != null ? chunk.a(x, y, z) : null;
 		}
 		return null;
@@ -129,7 +129,7 @@ public class NmsManager extends AbstractNmsManager {
 	@Override
 	public ReadOnlyChunk getReadOnlyChunk(World world, int chunkX, int chunkZ) {
 		ChunkProviderServer chunkProviderServer = world(world).getChunkProviderServer();
-		Chunk chunk = chunkProviderServer.getOrLoadChunkAt(chunkX, chunkZ);
+		Chunk chunk = chunkProviderServer.getChunkAt(chunkX, chunkZ);
 		return new ReadOnlyChunkWrapper(chunk);
 	}
 

--- a/orebfuscator-nms/orebfuscator-nms-v1_11_R1/src/main/java/net/imprex/orebfuscator/nms/v1_11_R1/NmsManager.java
+++ b/orebfuscator-nms/orebfuscator-nms-v1_11_R1/src/main/java/net/imprex/orebfuscator/nms/v1_11_R1/NmsManager.java
@@ -49,7 +49,7 @@ public class NmsManager extends AbstractNmsManager {
 
 		if (isChunkLoaded(worldServer, x >> 4, z >> 4) || loadChunk) {
 			// will load chunk if not loaded already
-			Chunk chunk = chunkProviderServer.getOrLoadChunkAt(x >> 4, z >> 4);
+			Chunk chunk = chunkProviderServer.getChunkAt(x >> 4, z >> 4);
 			return chunk != null ? chunk.a(x, y, z) : null;
 		}
 		return null;
@@ -128,7 +128,7 @@ public class NmsManager extends AbstractNmsManager {
 	@Override
 	public ReadOnlyChunk getReadOnlyChunk(World world, int chunkX, int chunkZ) {
 		ChunkProviderServer chunkProviderServer = world(world).getChunkProviderServer();
-		Chunk chunk = chunkProviderServer.getOrLoadChunkAt(chunkX, chunkZ);
+		Chunk chunk = chunkProviderServer.getChunkAt(chunkX, chunkZ);
 		return new ReadOnlyChunkWrapper(chunk);
 	}
 

--- a/orebfuscator-nms/orebfuscator-nms-v1_12_R1/src/main/java/net/imprex/orebfuscator/nms/v1_12_R1/NmsManager.java
+++ b/orebfuscator-nms/orebfuscator-nms-v1_12_R1/src/main/java/net/imprex/orebfuscator/nms/v1_12_R1/NmsManager.java
@@ -49,7 +49,7 @@ public class NmsManager extends AbstractNmsManager {
 
 		if (isChunkLoaded(worldServer, x >> 4, z >> 4) || loadChunk) {
 			// will load chunk if not loaded already
-			Chunk chunk = chunkProviderServer.getOrLoadChunkAt(x >> 4, z >> 4);
+			Chunk chunk = chunkProviderServer.getChunkAt(x >> 4, z >> 4);
 			return chunk != null ? chunk.a(x, y, z) : null;
 		}
 		return null;
@@ -128,7 +128,7 @@ public class NmsManager extends AbstractNmsManager {
 	@Override
 	public ReadOnlyChunk getReadOnlyChunk(World world, int chunkX, int chunkZ) {
 		ChunkProviderServer chunkProviderServer = world(world).getChunkProviderServer();
-		Chunk chunk = chunkProviderServer.getOrLoadChunkAt(chunkX, chunkZ);
+		Chunk chunk = chunkProviderServer.getChunkAt(chunkX, chunkZ);
 		return new ReadOnlyChunkWrapper(chunk);
 	}
 

--- a/orebfuscator-nms/orebfuscator-nms-v1_13_R1/src/main/java/net/imprex/orebfuscator/nms/v1_13_R1/NmsManager.java
+++ b/orebfuscator-nms/orebfuscator-nms-v1_13_R1/src/main/java/net/imprex/orebfuscator/nms/v1_13_R1/NmsManager.java
@@ -49,7 +49,7 @@ public class NmsManager extends AbstractNmsManager {
 
 		if (isChunkLoaded(worldServer, x >> 4, z >> 4) || loadChunk) {
 			// will load chunk if not loaded already
-			Chunk chunk = chunkProviderServer.getOrLoadChunkAt(x >> 4, z >> 4);
+			Chunk chunk = chunkProviderServer.getChunkAt(x >> 4, z >> 4);
 			return chunk != null ? chunk.getBlockData(x, y, z) : null;
 		}
 		return null;
@@ -127,7 +127,7 @@ public class NmsManager extends AbstractNmsManager {
 	@Override
 	public ReadOnlyChunk getReadOnlyChunk(World world, int chunkX, int chunkZ) {
 		ChunkProviderServer chunkProviderServer = world(world).getChunkProviderServer();
-		Chunk chunk = chunkProviderServer.getOrLoadChunkAt(chunkX, chunkZ);
+		Chunk chunk = chunkProviderServer.getChunkAt(chunkX, chunkZ);
 		return new ReadOnlyChunkWrapper(chunk);
 	}
 

--- a/orebfuscator-nms/orebfuscator-nms-v1_13_R2/src/main/java/net/imprex/orebfuscator/nms/v1_13_R2/NmsManager.java
+++ b/orebfuscator-nms/orebfuscator-nms-v1_13_R2/src/main/java/net/imprex/orebfuscator/nms/v1_13_R2/NmsManager.java
@@ -50,7 +50,7 @@ public class NmsManager extends AbstractNmsManager {
 
 		if (isChunkLoaded(worldServer, x >> 4, z >> 4) || loadChunk) {
 			// will load chunk if not loaded already
-			Chunk chunk = chunkProviderServer.getChunkAt(x >> 4, z >> 4, true, false);
+			Chunk chunk = chunkProviderServer.getChunkAt(x >> 4, z >> 4, true, true);
 			return chunk != null ? chunk.getBlockData(x, y, z) : null;
 		}
 		return null;
@@ -128,7 +128,7 @@ public class NmsManager extends AbstractNmsManager {
 	@Override
 	public ReadOnlyChunk getReadOnlyChunk(World world, int chunkX, int chunkZ) {
 		ChunkProviderServer chunkProviderServer = world(world).getChunkProvider();
-		Chunk chunk = chunkProviderServer.getChunkAt(chunkX, chunkZ, true, false);
+		Chunk chunk = chunkProviderServer.getChunkAt(chunkX, chunkZ, true, true);
 		return new ReadOnlyChunkWrapper(chunk);
 	}
 

--- a/orebfuscator-nms/orebfuscator-nms-v1_9_R2/src/main/java/net/imprex/orebfuscator/nms/v1_9_R2/NmsManager.java
+++ b/orebfuscator-nms/orebfuscator-nms-v1_9_R2/src/main/java/net/imprex/orebfuscator/nms/v1_9_R2/NmsManager.java
@@ -49,7 +49,7 @@ public class NmsManager extends AbstractNmsManager {
 
 		if (isChunkLoaded(worldServer, x >> 4, z >> 4) || loadChunk) {
 			// will load chunk if not loaded already
-			Chunk chunk = chunkProviderServer.getOrLoadChunkAt(x >> 4, z >> 4);
+			Chunk chunk = chunkProviderServer.getChunkAt(x >> 4, z >> 4);
 			return chunk != null ? chunk.a(x, y, z) : null;
 		}
 		return null;
@@ -128,7 +128,7 @@ public class NmsManager extends AbstractNmsManager {
 	@Override
 	public ReadOnlyChunk getReadOnlyChunk(World world, int chunkX, int chunkZ) {
 		ChunkProviderServer chunkProviderServer = world(world).getChunkProviderServer();
-		Chunk chunk = chunkProviderServer.getOrLoadChunkAt(chunkX, chunkZ);
+		Chunk chunk = chunkProviderServer.getChunkAt(chunkX, chunkZ);
 		return new ReadOnlyChunkWrapper(chunk);
 	}
 

--- a/orebfuscator-plugin/src/main/java/net/imprex/orebfuscator/Orebfuscator.java
+++ b/orebfuscator-plugin/src/main/java/net/imprex/orebfuscator/Orebfuscator.java
@@ -83,8 +83,7 @@ public class Orebfuscator extends JavaPlugin implements Listener {
 					new DefaultOrebfuscatorService(this),
 					this, ServicePriority.Normal);
 		} catch (Exception e) {
-			OFCLogger.log(Level.SEVERE, "An error occurred while enabling plugin");
-			OFCLogger.error(e);
+			OFCLogger.error("An error occurred while enabling plugin", e);
 
 			this.getServer().getPluginManager().registerEvent(PluginEnableEvent.class, this, EventPriority.NORMAL,
 					this::onEnableFailed, this);

--- a/orebfuscator-plugin/src/main/java/net/imprex/orebfuscator/Orebfuscator.java
+++ b/orebfuscator-plugin/src/main/java/net/imprex/orebfuscator/Orebfuscator.java
@@ -84,7 +84,7 @@ public class Orebfuscator extends JavaPlugin implements Listener {
 					this, ServicePriority.Normal);
 		} catch (Exception e) {
 			OFCLogger.log(Level.SEVERE, "An error occurred while enabling plugin");
-			OFCLogger.err(e);
+			OFCLogger.error(e);
 
 			this.getServer().getPluginManager().registerEvent(PluginEnableEvent.class, this, EventPriority.NORMAL,
 					this::onEnableFailed, this);

--- a/orebfuscator-plugin/src/main/java/net/imprex/orebfuscator/cache/ObfuscationCache.java
+++ b/orebfuscator-plugin/src/main/java/net/imprex/orebfuscator/cache/ObfuscationCache.java
@@ -1,7 +1,5 @@
 package net.imprex.orebfuscator.cache;
 
-import java.util.Objects;
-import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
 import java.util.concurrent.TimeUnit;
 

--- a/orebfuscator-plugin/src/main/java/net/imprex/orebfuscator/cache/ObfuscationCache.java
+++ b/orebfuscator-plugin/src/main/java/net/imprex/orebfuscator/cache/ObfuscationCache.java
@@ -50,19 +50,22 @@ public class ObfuscationCache {
 
 		ObfuscationResult cacheChunk = this.cache.getIfPresent(key);
 		if (request.isValid(cacheChunk)) {
-			return CompletableFuture.completedFuture(cacheChunk);
+			return request.complete(cacheChunk);
 		}
 
-		return this.serializer.read(key).thenCompose(diskChunk -> {
+		this.serializer.read(key).thenCompose(diskChunk -> {
 			if (request.isValid(diskChunk)) {
-				return CompletableFuture.completedFuture(diskChunk);
+				return request.complete(diskChunk);
+			} else {
+				return request.submitForObfuscation().exceptionally(throwable -> null);
 			}
-
-			return request.submitForObfuscation();
-		}).thenApply(chunk -> {
-			this.cache.put(key, Objects.requireNonNull(chunk));
-			return chunk;
+		}).thenAccept(chunk -> {
+			if (chunk != null) {
+				this.cache.put(key, chunk);
+			}
 		});
+
+		return request.getFuture();
 	}
 
 	public void invalidate(ChunkPosition key) {

--- a/orebfuscator-plugin/src/main/java/net/imprex/orebfuscator/obfuscation/ObfuscationListener.java
+++ b/orebfuscator-plugin/src/main/java/net/imprex/orebfuscator/obfuscation/ObfuscationListener.java
@@ -59,13 +59,13 @@ public class ObfuscationListener extends PacketAdapter {
 
 		this.obfuscationSystem.obfuscate(struct).whenComplete((chunk, throwable) -> {
 			if (throwable != null) {
-				this.completeExceptionally(event, throwable);
+				this.completeExceptionally(event, struct, throwable);
 			} else if (chunk != null) {
 				this.complete(event, struct, chunk);
 			} else {
 				this.skipChunkForProcessing(event);
-				OFCLogger.warn(String.format("skipping chunk[x=%d, z=%d] because obfuscation result is missing",
-						struct.chunkX, struct.chunkZ));
+				OFCLogger.warn(String.format("skipping chunk[world=%s, x=%d, z=%d] because obfuscation result is missing",
+						struct.world.getName(), struct.chunkX, struct.chunkZ));
 			}
 		});
 	}
@@ -82,8 +82,9 @@ public class ObfuscationListener extends PacketAdapter {
 		this.asynchronousManager.signalPacketTransmission(event);
 	}
 
-	private void completeExceptionally(PacketEvent event, Throwable throwable) {
-		throwable.printStackTrace();
+	private void completeExceptionally(PacketEvent event, ChunkStruct struct, Throwable throwable) {
+		OFCLogger.error(String.format("An error occurred while obfuscating chunk[world=%s, x=%d, z=%d]",
+				struct.world.getName(), struct.chunkX, struct.chunkZ), throwable);
 		this.skipChunkForProcessing(event);
 	}
 

--- a/orebfuscator-plugin/src/main/java/net/imprex/orebfuscator/obfuscation/ObfuscationRequest.java
+++ b/orebfuscator-plugin/src/main/java/net/imprex/orebfuscator/obfuscation/ObfuscationRequest.java
@@ -44,6 +44,10 @@ public class ObfuscationRequest {
 		this.chunkStruct = chunkStruct;
 	}
 
+	public CompletableFuture<ObfuscationResult> getFuture() {
+		return future;
+	}
+
 	public ChunkPosition getPosition() {
 		return position;
 	}
@@ -65,11 +69,13 @@ public class ObfuscationRequest {
 		return new ObfuscationResult(this.position, this.chunkHash, data, blockEntities, proximityBlocks);
 	}
 
-	public void complete(ObfuscationResult result) {
+	public CompletionStage<ObfuscationResult> complete(ObfuscationResult result) {
 		this.future.complete(result);
+		return this.future;
 	}
 
-	public void completeExceptionally(Throwable throwable) {
+	public CompletionStage<ObfuscationResult> completeExceptionally(Throwable throwable) {
 		this.future.completeExceptionally(throwable);
+		return this.future;
 	}
 }

--- a/orebfuscator-plugin/src/main/java/net/imprex/orebfuscator/util/OFCLogger.java
+++ b/orebfuscator-plugin/src/main/java/net/imprex/orebfuscator/util/OFCLogger.java
@@ -49,6 +49,13 @@ public class OFCLogger {
 	}
 
 	/**
+	 * Log an error
+	 */
+	public static void err(String message, Throwable e) {
+		log(Level.SEVERE, message, e);
+	}
+
+	/**
 	 * Log with a specified level and throwable
 	 */
 	public static void log(Level level, String message, Throwable throwable) {

--- a/orebfuscator-plugin/src/main/java/net/imprex/orebfuscator/util/OFCLogger.java
+++ b/orebfuscator-plugin/src/main/java/net/imprex/orebfuscator/util/OFCLogger.java
@@ -44,14 +44,14 @@ public class OFCLogger {
 	/**
 	 * Log an error
 	 */
-	public static void err(Throwable e) {
+	public static void error(Throwable e) {
 		log(Level.SEVERE, e.getMessage(), e);
 	}
 
 	/**
 	 * Log an error
 	 */
-	public static void err(String message, Throwable e) {
+	public static void error(String message, Throwable e) {
 		log(Level.SEVERE, message, e);
 	}
 

